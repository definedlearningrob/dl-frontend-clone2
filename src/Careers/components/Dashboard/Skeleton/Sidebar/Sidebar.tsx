import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
