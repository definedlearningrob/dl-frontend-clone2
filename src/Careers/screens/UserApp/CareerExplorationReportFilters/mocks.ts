import { MockedResponse } from '@apollo/client/testing';
import {
  CareerExplorationReportFiltersDocument,
  CareerExplorationReportSummaryDocument,
  CareerExplorationReportVisitCountsDocument,
} from '@graphql/dc/users/hooks';

import { TCareerExplorationReportFiltersData } from '@dc/graphql/user/queries/careerExplorationReportFilters';

export const careerExplorationReportSummaryMock = {
  request: {
    query: CareerExplorationReportSummaryDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          summary: {
            engagementsCount: 10,
            clusterSCount: 2,
            pathwaysCount: 5,
            studentCount: 12,
            usersCount: 3,
          },
        },
      },
    },
  },
};

export const visitCountsSpy = jest.fn();

export const careerExplorationReportVisitCountsMock = {
  request: {
    query: CareerExplorationReportVisitCountsDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result() {
    visitCountsSpy();

    return {
      data: {
        reports: {
          pathwayReport: [
            {
              visitsCount: 396,
              cluster: {
                name: 'Agriculture, Food & Natural Resources',
                id: '1',
                __typename: 'Cluster',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '5',
                    name: 'Natural Resources Systems',
                  },
                },
                {
                  visitsCount: 55,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '2',
                    name: 'Animal Systems',
                  },
                },
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '6',
                    name: 'Plant Systems',
                  },
                },
              ],
            },
            {
              visitsCount: 10,
              cluster: {
                name: 'Transportation, Distribution & Logistics',
                id: '11',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 2,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '52',
                    name: 'Facility and Mobile Equipment Maintenance',
                  },
                },
                {
                  visitsCount: 3,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '49',
                    name: 'Transportation Operations',
                  },
                },
                {
                  visitsCount: 5,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '51',
                    name: 'Warehousing & Distribution Center Operations',
                  },
                },
              ],
            },
          ],
        },
      },
    };
  },
};

export const careerExplorationReportFiltersMock: MockedResponse<TCareerExplorationReportFiltersData> =
  {
    request: {
      query: CareerExplorationReportFiltersDocument,
      variables: {
        filters: {},
        entityFilter: { nameCont: '' },
        schoolClassFilter: { nameCont: '' },
        userFilter: { fullNameCont: '' },
      },
    },
    result: {
      data: {
        pathwayReportFilters: {
          entities: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '1234-5678', name: 'City High School' }],
          },
          gradeLevels: ['10', '11', '12', 'Postsecondary'],
          users: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
          },
          schoolClasses: {
            nodesCount: 0,
            pagesCount: 0,
            nodes: [],
          },
        },
      },
    },
  };
