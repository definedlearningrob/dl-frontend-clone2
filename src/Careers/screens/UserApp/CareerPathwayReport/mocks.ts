import { MockedResponse } from '@apollo/client/testing';
import { CareerExplorationReportFullDataDocument } from '@graphql/dc/users/hooks';
import { CareerExplorationReportFullDataQuery } from '@graphql/dc/users/operations';

export const careerExplorationReportMock: MockedResponse<CareerExplorationReportFullDataQuery> = {
  request: {
    query: CareerExplorationReportFullDataDocument,
    variables: {
      filter: { schoolYear: 2023 },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          visitResults: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [
              {
                clusterNames: ['Education & Training'],
                isEnrolled: true,
                pathwayNames: ['Teaching/Training'],
                resourceId: '16',
                resourceName: 'Teaching/Training',
                resourceType: 'Pathway',
                visitorId: '1234',
                visitorName: 'Jeremiah Bell',
                visitorEmail: 'jeremiah.bell@example.com',
                visitorSisId: null,
                visitorType: 'Student',
                visitsCount: 5,
              },
            ],
          },
        },
      },
    },
  },
};
