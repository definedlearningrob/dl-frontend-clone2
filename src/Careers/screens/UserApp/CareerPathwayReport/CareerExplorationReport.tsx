import { useTranslation } from 'react-i18next';
import { useToggle } from 'react-use';
import { useEffect } from 'react';
import { useCareerExplorationReportSummaryQuery } from '@graphql/dc/users/hooks';

import { CareerExplorationFiltersModal } from '@dc/components/CareerPathwayReport/CareerExplorationFiltersModal';
import { CareerExplorationSummary } from '@dc/screens/UserApp/CareerPathwayReport/CareerExplorationSummary/CareerExplorationSummary';
import { CareerExplorationEngagement } from '@dc/screens/UserApp/CareerPathwayReport/CareerPathwayEngagementsChart/CareerExplorationEngagement';
import { useCareerExplorationReportFilters } from '@dc/components/CareerPathwayReport/useCareerExplorationReportFilters';
import { CareerExplorationFullData } from '@dc/screens/UserApp/CareerPathwayReport/CareerExplorationTable/CareerExplorationFullData';

import Button from '@shared/components/Button/Button';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';
import { ReactComponent as FilterIcon } from '@shared/svg/filter_outlined.svg';
import { MainContent } from '@shared/components/MainContent/MainContent';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';
import Card from '@shared/components/Card/Card';
import { FilterProvider } from '@shared/components/FilterProvider/FilterProvider';

export const CareerExplorationReport = () => {
  const { t } = useTranslation();
  const { setBackNavButton } = useNavigation();
  const { variables } = useCareerExplorationReportFilters();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });
  const [isFiltersModalOpen, toggleFiltersModalOpen] = useToggle(false);

  const {
    data: summaryData,
    loading,
    error,
  } = useCareerExplorationReportSummaryQuery({
    variables,
    fetchPolicy: 'no-cache',
  });

  useEffect(() => {
    setBackNavButton(true);

    return () => setBackNavButton(false);
  }, []);

  if (error) {
    return <div className='text-center'>{t('common.messages.dataLoadingError')}</div>;
  }

  const summary = summaryData?.reports?.pathwayReport?.summary;

  return (
    <MainContent>
      <div className='pb-xs mb-xs xxxl:mb-sm sticky top-lg z-high bg-neutral-200'>
        <div className='flex justify-between items-center'>
          <h4 className='text-base xxxl:text-lg mb-0'>
            {t('reports.reportName', {
              reportName: t('reports.reportTitle.careerPathway'),
            })}
          </h4>
          <div className='flex gap-sm'>
            <Button
              Icon={FilterIcon}
              size={isFullHD ? 'md' : 'sm'}
              variant='primary'
              onClick={toggleFiltersModalOpen}>
              {t('reports.changeFilters')}
            </Button>
          </div>
        </div>
        <CareerExplorationFiltersModal
          isOpen={isFiltersModalOpen}
          onClose={() => {
            toggleFiltersModalOpen(false);
          }}
        />
      </div>
      <div className='flex flex-col gap-base xxxl:gap-md'>
        <Card>
          <CareerExplorationSummary
            isLoading={loading}
            studentsTotal={summary?.studentsCount}
            summary={summary}
          />
        </Card>
        <Card>
          <CareerExplorationEngagement />
        </Card>
        <Card>
          <FilterProvider initialFilters={{ fullNameOrSisIdCont: '', pathwayIdIn: [] }}>
            <CareerExplorationFullData />
          </FilterProvider>
        </Card>
      </div>
    </MainContent>
  );
};
