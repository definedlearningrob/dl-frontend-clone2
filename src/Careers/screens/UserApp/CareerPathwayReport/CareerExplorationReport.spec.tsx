import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen, waitFor } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import {
  GenerateCareerExplorationReportDocument,
  CareerExplorationReportDocument,
  CareerExplorationReportQueryResult,
} from '@graphql/dc/users/hooks';
import {
  GenerateCareerExplorationReportMutationVariables,
  GenerateCareerExplorationReportMutation,
} from '@graphql/dc/users/operations';

import { renderWithRouter } from '@dc/utils/test';
import { careerExplorationReportMock } from '@dc/screens/UserApp/CareerPathwayReport/mocks';
import { CareerExplorationReport } from '@dc/screens/UserApp/CareerPathwayReport/CareerExplorationReport';
import {
  careerExplorationReportFiltersMock,
  careerExplorationReportSummaryMock,
  careerExplorationReportVisitCountsMock,
} from '@dc/screens/UserApp/CareerExplorationReportFilters/mocks';

import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
import { ReportFiltersProvider } from '@shared/components/ReportFiltersProvider/ReportFiltersProvider';
import { ALL_OPTION } from '@shared/components/MultiSelect';
import { FileDownloadProvider } from '@shared/hooks/useFileDownload';

const generateReportSpy = jest.fn();

const reportFilters = {
  schoolYear: { value: 2023, label: '2023 / 2024' },
  entities: [ALL_OPTION],
  gradeLevels: [ALL_OPTION],
  users: [ALL_OPTION],
  schoolClasses: [ALL_OPTION],
  entityFilter: '',
  userFilter: '',
  schoolClassFilter: '',
};

const generateCareerExplorationReportMock: MockedResponse<
  GenerateCareerExplorationReportMutation,
  GenerateCareerExplorationReportMutationVariables
> = {
  request: {
    query: GenerateCareerExplorationReportDocument,
    variables: {
      input: {
        schoolYear: 2023,
      },
    },
  },
  result() {
    generateReportSpy();

    return {
      data: {
        generatePathwayReport: {
          pathwayReport: {
            id: '1234-5678',
          },
        },
      },
    };
  },
};

const renderComponent = (mocks: MockedResponse[]) =>
  renderWithRouter(
    <MockedProvider mocks={mocks}>
      <ReportFiltersProvider initialFilters={reportFilters}>
        <FileDownloadProvider>
          <NavigationContextProvider>
            <CareerExplorationReport />
          </NavigationContextProvider>
        </FileDownloadProvider>
      </ReportFiltersProvider>
    </MockedProvider>
  );

describe('CareerExplorationReport', () => {
  it('should render correctly', async () => {
    const { container } = renderComponent([
      careerExplorationReportSummaryMock,
      careerExplorationReportSummaryMock,
      careerExplorationReportVisitCountsMock,
      careerExplorationReportVisitCountsMock,
    ]);

    expect(
      await screen.findByRole('heading', { name: 'Career Exploration Report' })
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should open filters modal after clicking on the button', async () => {
    renderComponent([careerExplorationReportSummaryMock, careerExplorationReportFiltersMock]);

    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();

    userEvent.click(screen.getByRole('button', { name: 'Change filters' }));

    expect(await screen.findByRole('dialog')).toBeInTheDocument();

    const schoolsDropdown = await screen.findByRole('combobox', { name: 'Schools All' });

    expect(schoolsDropdown).toBeInTheDocument();
    expect(screen.getByRole('combobox', { name: 'Grade Levels All' })).toBeInTheDocument();
    expect(screen.getByRole('combobox', { name: 'Teachers All' })).toBeInTheDocument();
    expect(screen.getByRole('combobox', { name: 'School Year 2023 / 2024' })).toBeInTheDocument();
    expect(screen.getByTestId('report-filters-select-schoolClasses')).toHaveTextContent('All');
  });

  it('should display error message if there is an error', async () => {
    const careerExplorationReportErrorMock: MockedResponse<CareerExplorationReportQueryResult> = {
      request: {
        query: CareerExplorationReportDocument,
        variables: {
          filter: { schoolYear: 2023 },
        },
      },
      error: new Error('An error occured'),
    };

    renderComponent([careerExplorationReportErrorMock]);

    expect(
      await screen.findByText('An error occurred while loading the data.')
    ).toBeInTheDocument();
  });

  it('should generate report after clicking on the button', async () => {
    renderComponent([careerExplorationReportMock, generateCareerExplorationReportMock]);

    userEvent.click(screen.getByRole('button', { name: 'Download CSV' }));

    await waitFor(() => expect(generateReportSpy).toHaveBeenCalledTimes(1));
  });
});
