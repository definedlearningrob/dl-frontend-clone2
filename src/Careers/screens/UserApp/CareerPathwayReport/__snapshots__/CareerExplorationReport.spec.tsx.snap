// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareerExplorationReport should render correctly 1`] = `
<div>
  <div
    class="p-base xxxl:p-md"
    id="read"
  >
    <div
      class="pb-xs mb-xs xxxl:mb-sm sticky top-lg z-high bg-neutral-200"
    >
      <div
        class="flex justify-between items-center"
      >
        <h4
          class="text-base xxxl:text-lg mb-0"
        >
          Career Exploration Report
        </h4>
        <div
          class="flex gap-sm"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            type="button"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 h-[14px] mt-[-1px]"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="items-center flex justify-center gap-xs"
            >
              Change filters
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-base xxxl:gap-md"
    >
      <div
        class="card"
      >
        <div>
          <h5
            class="mb-xs xxxl:mb-sm text-sm xxxl:text-base"
          >
            Summary
          </h5>
          <div
            class="mb-sm xxxl:mb-base text-xs xxxl:text-sm leading-lg"
          >
            Below you can see an overview of 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class="text-primary-500 font-medium cursor-default"
              >
                all grade levels
              </span>
            </div>
             including 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class=""
              >
                0 students
              </span>
            </div>
             in 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class=""
              >
                2023/2024
              </span>
            </div>
            , with a specific focus on 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class="text-primary-500 font-medium cursor-default"
              >
                all teachers
              </span>
            </div>
            , 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class="text-primary-500 font-medium cursor-default"
              >
                all schools
              </span>
            </div>
             and 
            <div
              class="inline"
              data-state="closed"
            >
              <span
                class="text-primary-500 font-medium cursor-default"
              >
                all classes
              </span>
            </div>
            .
          </div>
          <h5
            class="mb-xs xxxl:mb-sm text-sm xxxl:text-base"
          >
            Statistics
          </h5>
          <p
            class="mb-xs xxxl:mb-sm text-xs xxxl:text-sm leading-lg flex items-center"
          >
            Overview of student engagement metrics within the career pathways and clusters.
          </p>
          <div
            class="flex w-full justify-evenly rounded-sm border border-neutral-300 [&_div]:border-r [&_div]:border-neutral-300 [&_div:last-child]:border-none overflow-hidden"
          >
            <div
              class="flex-1"
            >
              <h6
                class="p-sm text-xxs xxxl:text-xs font-bold bg-neutral-200 mb-0 whitespace-nowrap border-b border-b-neutral-300"
              >
                Total Engagements
              </h6>
              <div
                class="bg-white px-sm py-xs text-xs xxxl:text-sm text-center font-medium"
              >
                --
              </div>
            </div>
            <div
              class="flex-1"
            >
              <h6
                class="p-sm text-xxs xxxl:text-xs font-bold bg-neutral-200 mb-0 whitespace-nowrap border-b border-b-neutral-300"
              >
                Career Clusters
              </h6>
              <div
                class="bg-white px-sm py-xs text-xs xxxl:text-sm text-center font-medium"
              >
                --
              </div>
            </div>
            <div
              class="flex-1"
            >
              <h6
                class="p-sm text-xxs xxxl:text-xs font-bold bg-neutral-200 mb-0 whitespace-nowrap border-b border-b-neutral-300"
              >
                Pathways
              </h6>
              <div
                class="bg-white px-sm py-xs text-xs xxxl:text-sm text-center font-medium"
              >
                --
              </div>
            </div>
            <div
              class="flex-1"
            >
              <h6
                class="p-sm text-xxs xxxl:text-xs font-bold bg-neutral-200 mb-0 whitespace-nowrap border-b border-b-neutral-300"
              >
                Educators
              </h6>
              <div
                class="bg-white px-sm py-xs text-xs xxxl:text-sm text-center font-medium"
              >
                --
              </div>
            </div>
            <div
              class="flex-1"
            >
              <h6
                class="p-sm text-xxs xxxl:text-xs font-bold bg-neutral-200 mb-0 whitespace-nowrap border-b border-b-neutral-300"
              >
                Students
              </h6>
              <div
                class="bg-white px-sm py-xs text-xs xxxl:text-sm text-center font-medium"
              >
                --
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <h5
          class="text-sm xxxl:text-base mb-xs xxxl:mb-sm"
        >
          Experiences
        </h5>
        <p
          class="text-xs xxxl:text-sm leading-lg xxxl:mb-base"
        >
          Number of experiences grouped by clusters
        </p>
        <div
          class="p-md"
        >
          <div
            class="flex flex-col gap-xxs xxxl:gap-sm"
          >
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full w-5/6"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full w-4/6"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full w-3/6"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full w-2/6"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
            <div
              class="flex gap-sm h-[32px] xxxl:h-[40px]"
            >
              <div
                class="basis-[320px] shrink-0 flex items-center"
              >
                <div
                  class="flex flex-col ml-auto gap-xxs xxxl:gap-sm w-3/4"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle radius-md height-auto width-full-width darker animated"
                  />
                </div>
              </div>
              <div
                class="grow"
              >
                <div
                  class="h-full w-1/6"
                >
                  <div
                    class="skeleton-rectangle skeletonRectangle !rounded-xxs radius-md height-full-height width-full-width darker animated"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="mb-sm xxxl:mb-base flex justify-between items-end"
        >
          <div
            class="flex flex-col gap-xs"
          >
            <h5
              class="inline me-xs text-sm xxxl:text-base mb-0"
            >
              Full Data 
              <span
                class="text-neutral-600"
              >
                (0 results)
              </span>
            </h5>
            <p
              class="text-neutral-700 text-xs xxxl:text-sm mb-0"
            >
              Below you can see all collected data about engagements.
            </p>
          </div>
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
            data-testid="button"
            type="button"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 h-[14px] mt-[-1px]"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="items-center flex justify-center gap-xs"
            >
              Download CSV
            </span>
          </button>
        </div>
        <div
          class="border border-neutral-300 rounded-sm mb-base"
        >
          <div
            class="flex gap-sm items-center bg-neutral-200 px-sm py-xs xxxl:px-base xxxl:py-x rounded-t-sm"
          >
            <label
              class="flex flex-col items-start text-font-primary w-full !w-[300px] xxxl:!w-[320px] gap-xxxs"
            >
              <div
                class="w-full flex gap-xs items-center justify-between bg-white rounded-sm border border-neutral-300 hover:border-neutral-400 group focus-within:!border-primary-500 p-xs flex-row-reverse"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 h-[14px]"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  class="flex gap-xxs grow"
                >
                  <input
                    class="bg-transparent text-font-primary min-w-0 w-full grow w-full outline-none transition-colors autofill:shadow-[inset_0_0_0px_1000px_rgb(252,252,252)] placeholder:text-font-secondary leading-[14px] text-xxs"
                    placeholder="Search by user name or user ID"
                    type="text"
                  />
                </span>
              </div>
            </label>
            <label
              class="flex flex-col leading-lg min-w-[120px] gap-xxxs"
              data-testid="clusters-select"
            >
              
              <div
                class="min-w-[240px] css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 !bg-white select__control css-13cymwt-control"
                >
                  <div
                    class="flex gap-xxs items-center !flex-nowrap !text-xxs !p-xs !pr-0 !py-xxs h-[30px] select__value-container select__value-container--is-multi css-1fdsijx-ValueContainer"
                  >
                    <div
                      class="!text-font-secondary !leading-[22px] h-[22px] select__placeholder css-1jqq78o-placeholder"
                      id="react-select-2-placeholder"
                    >
                      Search by clusters or pathways
                    </div>
                    <div
                      class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-qbdosj-Input"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="select__input"
                        id="select-input-clusterAndPathway"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                      class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 text-neutral-400"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <input
                    name="clusterAndPathway"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </label>
          </div>
          <div
            class="border border-neutral-300 rounded-b-sm h-[512px] xxxl:h-[576px] overflow-auto"
          >
            <div
              class="overflow-hidden flex flex-col h-full min-h-0"
            >
              <div
                class="grow z-lowest scrollbar"
              >
                <table
                  class="w-full table-fixed overflow-auto"
                >
                  <thead
                    class="sticky top-0 bg-white z-lower tableHead"
                  >
                    <tr>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 60px;"
                      >
                        <div
                          class=""
                        >
                          ID
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 80px;"
                      >
                        <div
                          class=""
                        >
                          Name
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 60px;"
                      >
                        <div
                          class=""
                        >
                          Email
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 60px;"
                      >
                        <div
                          class=""
                        >
                          Visitor Type
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 120px;"
                      >
                        <div
                          class=""
                        >
                          Resource Name
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 80px;"
                      >
                        <div
                          class=""
                        >
                          Resource Type
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 150px;"
                      >
                        <div
                          class=""
                        >
                          Clusters
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 80px;"
                      >
                        <div
                          class=""
                        >
                          Pathways
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 60px;"
                      >
                        <div
                          class=""
                        >
                          Enrolled
                        </div>
                      </th>
                      <th
                        class="text-xxs xxxl:text-xs font-medium text-left leading-lg p-xs xxxl:px-x first:pl-base last:pr-base xxxl:first:pl-md xxxl:last:pr-md"
                        colspan="1"
                        style="width: 60px;"
                      >
                        <div
                          class=""
                        >
                          Visits Count
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody />
                </table>
                <div
                  class="text-center py-base text-xs"
                >
                  No results found
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
